src/jekyll/jk_document.rs

```diff
@@ -0,0 +1,285 @@
+            .merge_categories(custom_attributes
+                                  .remove("categories")
+                                  .and_then(|v| {
+                                                v.as_array()
+                                                    .map(|v| {
+                                                             v.iter()
+                                                                 .map(|v| v.to_string())
+                                                                 .collect()
+                                                         })
+                                            }))
+            .merge_permalink(custom_attributes
+                                 .remove("permalink")
+                                 .and_then(|v| v.as_str().map(|s| s.to_owned())))
+            .merge_draft(custom_attributes
+                             .remove("published")
+                             .and_then(|v| v.as_bool().map(|b| !b)))
+            .merge_layout(custom_attributes
+                              .remove("layout")
+                              .and_then(|v| v.as_str().map(|s| s.to_owned())))
+            .merge_published_date(custom_attributes
+                                      .remove("date")
+                                      .and_then(|d| {
+                                                    d.as_str().and_then(datetime::DateTime::parse)
+                                                }))
+            .merge_custom(custom_attributes)
+    }
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct JkDocument {
+    pub front: Option<String>,
+    pub content: Option<String>,
+}
+
+impl JkDocument {
+    pub fn parse_string(doc: String) -> Result<JkDocument> {
+        let (front, content) = split_document(&doc)?;
+        if front != None {
+            Err(ErrorKind::MissingFrontStart.into())
+        } else {
+            match split_document(content)? {
+                (None, _) => Err(ErrorKind::MissingFrontmatter.into()),
+                (Some(front), content) => {
+                    Ok(JkDocument {
+                           front: Some(front.to_owned()),
+                           content: Some(content.to_owned()),
+                       })
+                }
+            }
+        }
+    }
+
+    pub fn parse(source_file: &Path) -> Result<JkDocument> {
+        let doc: String = read_file(source_file)?;
+        JkDocument::parse_string(doc)
+    }
+
+    pub fn convert_front(front: Option<String>) -> Result<String> {
+        let front_value: JkFrontmatterBuilder = front
+            .map(|s| serde_yaml::from_str(&s))
+            .map_or(Ok(None), |r| r.map(Some))?
+            .unwrap_or_else(JkFrontmatterBuilder::new);
+
+        let front_builder: FrontmatterBuilder = front_value.into();
+        let front = front_builder.build()?;
+        let mut converted = serde_yaml::to_string(&front)?;
+        converted.drain(..4);
```

Comment:
Whats the `drain` doing?

Reply:
remove from self the range and return it: https://doc.rust-lang.org/std/string/struct.String.html#method.drain

Reply:
Let me clarify, what is it you're trying to remove?

Reply:
I'm removing the first "---" from Jekyll's front as it seems that Cobalt's one does not start with "---"

Reply:
Ah, makes sense.  Thanks for clarifying that.

Might be good to have a comment here to explain intent.  Maybe even put it in a function?  

```rust
fn dump_frontmatter(var: FrontMatter) -> Result<String> {
    let mut converted = serde_yaml::to_string(&front)?;
    // Remove yaml's leading `---`
    converted.drain(..4);
    Ok(converted)
}
```